Archive 2021 ideas list (#781)
mattermost/mattermost-developer-documentation: site/content/contribute/gsoc/_index.md

---
title: "Google Summer of Code"
description: "Ideas list for Google Summer of Code"
section: "contribute"
---

> # 2021 ideas

Reviewer: ```suggestion
# GSoC 2021 Ideas
```

> **Note**: Mattermost was **not** selected as an organization for Google Summer of Code 2021. All that follows is kept here for historical reasons. Hopefully, we will be able to use some of the ideas in 2022.

Reviewer: ```suggestion
**Note**: Mattermost was **not** selected as an organization for Google Summer of Code 2021. All that follows is kept here for historical reasons. Hopefully, we'll be able to use some of the ideas in 2022.
```

Make sure to take a look at [the full Google Summer of Code timeline](https://summerofcode.withgoogle.com/how-it-works/#timeline) for more details!

## Ideas list

### Core features

> All ideas in the _Core features_ category propose direct changes to the Mattermost product, either within the server, or within the web client.

Reviewer: ```suggestion
All ideas in the *Core features* category propose direct changes to the Mattermost product, either within the server, or within the web client.
```

-   **Difficulty**: Easy
-   **Mentors**: Daniel Espino García, Alejandro García Montoro

> #### Add the concept of _Importance_ to messages

Reviewer: ```suggestion
#### Add the concept of *Importance* to messages
```
